### PR TITLE
feat: seperate button to handle discovery by IP, cleaned up code

### PR DESCRIPTION
**Pull request by kovapatrik**
Added minimal modification of code. I think it's more intuitive like this.

### CODE REVIEW

File: src/platform.ts [hunk 1]
```diff
@@ -80,14 +80,16 @@ export class MideaPlatform implements DynamicPlatformPlugin {
...
-    Object.values(this.platformConfig.devices).forEach((device) => {
-      // for some reason, assigning the regex has to be inside the loop, else fails after first pass.
-      const regexIPv4 = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/gi;
-      const ip = device.advanced_options?.ip?.toString().trim();
-      if (regexIPv4.test(ip)) {
-        this.discover.discoverDeviceByIP(ip);
-      } else if (ip) {
-        this.log.warn(`[${device.name}] Invalid IP address in configuration: ${ip}`);
+    this.platformConfig.devices.forEach((device) => {
+      if (device.advanced_options.ip.length > 0) {
```
Comment (dkerr64) [inline src/platform.ts:86]: We don't need to test for zero length, regex will fail.  And we need to handle undefined advanced_options and ip

Comment (kovapatrik) [reply src/platform.ts:86]: They can't be undefined as they have default values.

Comment (dkerr64) [reply src/platform.ts:86]: > They can't be undefined as they have default values.

In theory.  But I can tell you it crashed for me during testing (my version, when I was working on it).  Remember that for the Homebridge config we have stripped out settings that are unchanged from default.

Comment (kovapatrik) [reply src/platform.ts:86]: The length check is needed, because the regex will fail for every device which has the default IP value (an empty string), thus logging it out.

Comment (kovapatrik) [reply src/platform.ts:86]: > The length check is needed, because the regex will fail for every device which has the default IP value (an empty string), thus logging it out.

No, I'm wrong here, you have added the test of the ip varible in the else case. I think my solution is much cleaner: why would we test for regex when we know that the string is an empty string?

Comment (dkerr64) [reply src/platform.ts:86]: Yes the regex will fail, so no need to do discovery by IP.  But it is not logged because I test for non-empty.

Comment (dkerr64) [reply src/platform.ts:86]: > > The length check is needed, because the regex will fail for every device which has the default IP value (an empty string), thus logging it out.
> 
> No, I'm wrong here, you have added the test of the ip varible in the else case. I think my solution is much cleaner: why would we test for regex when we know that the string is an empty string?

It was an extra sanity check.  But I'm fine with removing the regex and just accepting anything in the entry field and letting an error throw somewhere else if that is your preference.

Comment (kovapatrik) [reply src/platform.ts:86]: > We don't need to test for zero length, regex will fail. And we need to handle undefined advanced_options and ip

Yeah you are right, I merge the default config to the device config later in the code.

Comment (kovapatrik) [reply src/platform.ts:86]: I'm okay with the regex, I'm just saying I wouldn't test for it, when the ip is empty.

File: src/platform.ts [hunk 2]
```diff
@@ -81,13 +82,14 @@ export class MideaPlatform implements DynamicPlatformPlugin {
     this.log.info('Start device discovery...');
     // If IP address is in config then probe them directly
     this.platformConfig.devices.forEach((device) => {
+      device = defaultsDeep(device, defaultDeviceConfig);
```
Comment (dkerr64) [inline src/platform.ts:85]: Why bother?

Comment (kovapatrik) [reply src/platform.ts:85]: The question mark handling of advanced_options and ip isn't very describing because in the DeviceConfig type those fields are not optional.
